src/google/protobuf/util/internal/utility.cc

```diff
@@ -302,7 +302,7 @@ bool IsMap(const google::protobuf::Field& field,
 string DoubleAsString(double value) {
   if (value == std::numeric_limits<double>::infinity()) return "Infinity";
   if (value == -std::numeric_limits<double>::infinity()) return "-Infinity";
-  if (::isnan(value)) return "NaN";
+  if (std::isnan(value)) return "NaN";
```

Comment:
Hello Feng Xiao,

I guess in MSYS2 with MinGW-w64 latest development environment (gcc 4.9.2)
C++11 is the default and this module fails to compile.  What is your
standard way of dealing with such portability issues?  Would some sort of
#ifdef __cplusplus be fine?

Best regards,

Karol

On 29 June 2015 at 20:20, Feng Xiao <notifications@github.com> wrote:

> In src/google/protobuf/util/internal/utility.cc
> <https://github.com/google/protobuf/pull/551#discussion_r33494650>:
>
> > @@ -302,7 +302,7 @@ bool IsMap(const google::protobuf::Field& field,
> >  string DoubleAsString(double value) {
> >    if (value == std::numeric_limits<double>::infinity()) return "Infinity";
> >    if (value == -std::numeric_limits<double>::infinity()) return "-Infinity";
> > -  if (::isnan(value)) return "NaN";
> > +  if (std::isnan(value)) return "NaN";
>
> std::isnan is a C++11 feature. We probably need to stick with ::isnan. Why
> can't "::isnan" be used here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/protobuf/pull/551/files#r33494650>.
>